### DISCUSSION TITLE
How to work CONFIG_LOG_OVERRIDE_LEVEL?

### DISCUSSION THREAD

**Discussion by ValNyrov** [zephyrproject-rtos/zephyr#67157] [General]
Hi all!
My board is NRF52840-DK and Zephyr version is 3.5.99. Help me understand how CONFIG_LOG_OVERRIDE_LEVEL works. 
The help says:
**"Forces a minimum log level for all modules. Modules use their specified level if it is greater than this option, otherwise they use the level specified by this option instead of their default or whatever was manually set."**
My very simple program:
```
#include <zephyr/kernel.h>
#include <stdio.h>
#include <zephyr/sys/printk.h>
#include <zephyr/devicetree.h>
#include <zephyr/drivers/gpio.h>
#include <zephyr/logging/log.h>
#include <zephyr/usb/usb_device.h>
#include <zephyr/drivers/uart.h>

LOG_MODULE_REGISTER(main_func, CONFIG_LOG_DEFAULT_LEVEL);
int main(void)
{

	LOG_INF("Starting main thread...");

       LOG_INF("start main loop...");
	while(1){

		LOG_INF("delay 5s ...");
		k_sleep(K_MSEC(5000));
	}
	return 0;
}
```
and my prj.conf:
```
CONFIG_LOG=y
CONFIG_LOG_BACKEND_UART=y
CONFIG_LOG_MODE_OVERFLOW=n
CONFIG_LOG_BUFFER_SIZE=4096
CONFIG_LOG_BACKEND_UART_BUFFER_SIZE=64
CONFIG_LOG_DEFAULT_LEVEL=3
CONFIG_LOG_OVERRIDE_LEVEL=1
CONFIG_LOG_MAX_LEVEL=3
CONFIG_LOG_OUTPUT_FORMAT_LINUX_TIMESTAMP=y
```
That is, in my config CONFIG_LOG_DEFAULT_LEVEL > CONFIG_LOG_OVERRIDE_LEVEL and there should be output to the log, but this does not happen.
But if CONFIG_LOG_OVERRIDE_LEVEL=0 or 2 or 3 then there is output to the log. 
Is this a bug or am I missing something?

BR,
Valeriy

**Comment by ndrs-pst** [comment]
@ValNyrov, it looks like the issue with CONFIG_LOG_OVERRIDE_LEVEL has also been addressed in this https://github.com/zephyrproject-rtos/zephyr/issues/44029, though with a focus on the reverse logic (CONFIG_LOG_DEFAULT_LEVEL < CONFIG_LOG_OVERRIDE_LEVEL).
